src/Spectre.libStatistics/Statistics.h

```diff
@@ -48,6 +50,22 @@ constexpr DataType Mean(gsl::span<const DataType> data)
     return Sum(data) / (data.size() != 0 ? data.size() : 1);
 }
 
+/// <summary>
+/// Obtain median of the specified data.
+/// </summary>
+/// <param name="data">The data.</param>
+/// <returns>Median of the elements.</returns>
+template <class DataType>
+DataType Median(gsl::span<const DataType> data)
+{
+    static_assert(std::is_arithmetic_v<DataType>, "DataType: expected arithmetic.");
+    std::vector<std::remove_const<DataType>::type> sortedData(data.begin(), data.end());
+    std::sort(sortedData.begin(), sortedData.end());
+    if (data.size() == 0) throw spectre::core::exception::EmptyArgumentException("Median: Provided vector is empty");
```

Comment:
How about checking this before sorting operation?

Reply:
Good point actually :smile: